examples/example-react-task/src/pages/LoginPage.tsx

```diff
@@ -34,7 +34,7 @@ export function LoginPage() {
     if (result.operation === AuthOperationName.Register && result.success) {
       logIn({ email, password });
     }
-  }, [result.operation, result.success]);
+  }, [result.operation, result.success, logIn, email, password]);
```

Comment:
React dependency list warnings are not pretty 😉 so I understand the addition of these. Although, functions like `logIn()` will not change behavior during runtime, but as part of rerenders elsewhere in the app its reference may change, causing `useEffect()` to run unnecessarily. Similarly, `email` and `password` would cause `useEffect()` to run on each character entered. If only running it when `result.operation` or `result.success` changes, it minimizes those calls. (That's also why just `result` itelf is not added.)

Did you experience any faulty behavior from not having those dependencies listed or only warnings?

If you think it's better for the example app to run without those warnings, then I'm fine with adding to the dependency list 👍

Reply:
I agree that as we have it set up we don't _actually_ need these additional dependencies - the app functions correctly either way. However in general we should declare exhaustive dependencies. This is a super lightweight effect function so I think the impact of running the effect on each character input would be negligible. It has the added benefit of the app having a clean build with no errors/warnings out of the box which I think gives devs a lot more confidence.

Reply:
Sounds good 🙂